python/ray/serialization.py

```diff
@@ -14,17 +14,14 @@
                             ObjectReconstructionFailedMaxAttemptsExceededError,
                             ObjectReconstructionFailedLineageEvictedError,
                             RaySystemError, RuntimeEnvSetupError)
-from ray._raylet import (
-    split_buffer,
-    unpack_pickle5_buffers,
-    Pickle5Writer,
-    Pickle5SerializedObject,
-    MessagePackSerializer,
-    MessagePackSerializedObject,
-    RawSerializedObject,
-)
+from ray._raylet import (split_buffer, unpack_pickle5_buffers, Pickle5Writer,
+                         Pickle5SerializedObject, MessagePackSerializer,
+                         MessagePackSerializedObject, RawSerializedObject,
+                         ArrowSerializedObject)
 from ray import serialization_addons
 
+import pyarrow as pa
```

Comment:
Pyarrow isn't a base dependency right? So we can't import it at the top level here. We might need to detect pyarrow tables in some other way like inspecting the type name.